alica_engine/include/engine/AlicaTimer.h

```diff
@@ -0,0 +1,112 @@
+#pragma once
+
+#include "engine/AlicaClock.h"
+#include "engine/IAlicaTimer.h"
+
+#include <atomic>
+#include <condition_variable>
+#include <functional>
+#include <memory>
+#include <stdexcept>
+#include <thread>
+
+namespace alicaTimer
+{
+
+class SyncStopTimerImpl : public std::enable_shared_from_this<SyncStopTimerImpl>
+{
+    using Base = std::enable_shared_from_this<SyncStopTimerImpl>;
+
+public:
+    using TimerCb = std::function<void()>;
+
+    SyncStopTimerImpl(TimerCb&& userCb, alica::AlicaTime period)
+            : _userCb(std::move(userCb))
+            , _period(period.inMilliseconds())
+            , _active(false)
+    {
+    }
+
+    ~SyncStopTimerImpl()
+    {
+        if (_active.load()) {
+            stop();
+        }
+    }
+
+    void start()
+    {
+
+        if (_active.load()) {
+            stop();
+        }
+        _active.store(true);
+        _thread = std::thread([this]() {
+            while (_active.load()) {
+                _userCb();
+                std::this_thread::sleep_for(std::chrono::milliseconds(_period));
```

Comment:
This could cause problems shutting down with a long period.  Maybe we can set a max sleep time (say 0.5s) and sleeping in intervals of that much, checking if the thread should still be active in the meantime

Reply:
Done